products/views.py

```diff
@@ -19,4 +19,25 @@ def get(self, request, category_id):
                     'image_url'  : subcategory.image_url,
                     'category_id': category_id } for subcategory in subcategories]
         
-        return JsonResponse({'result':result}, status=200)    
\ No newline at end of file
+        return JsonResponse({'result':result}, status=200)    
+     
+class ProductDetailView(View):
+    def get(self, request, product_id):
+
+        try:
+            product = Product.objects.get(id=product_id)   
+            options = product.productoption_set.filter(product_id = product.id)  
+            
+            result = { 'id'              : product.id,
+                        'name'           : product.name,
+                        'number'         : product.number,
+                        'description'    : product.description,
+                        'image_url'      : product.image_url,
+                        'sub_category_id': product.sub_category_id,
+                        'options'        : [{ 'size':option.size.name,
+                                             'price':option.price } for option in options]}
```

Comment:
이런식으로 바꾸면 조금 더 깔끔 할 것 같습니다.
```suggestion
            result = { 
                'id'             : product.id,
                'name'           : product.name,
                'number'         : product.number,
                'description'    : product.description,
                'image_url'      : product.image_url,
                'sub_category_id': product.sub_category_id,
                'options'        : [{ 
                    'size'  : option.size.name,
                    'price' : option.price 
                } for option in options]
            }
```

Reply:
앗 넵 반영했습니다!!!!